testing/src/test/java/org/oppia/android/testing/espresso/TextInputActionTest.kt

```diff
@@ -0,0 +1,110 @@
+package org.oppia.android.testing.espresso
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.android.material.textfield.TextInputLayout
+import com.google.common.truth.Truth.assertThat
+import org.hamcrest.Description
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.testing.R
+import org.oppia.android.testing.TextInputActionTestActivity
+import org.oppia.android.testing.espresso.TextInputAction.Companion.ErrorTextExisted
+import org.oppia.android.testing.espresso.TextInputAction.Companion.ErrorTextNotExisted
+import org.robolectric.annotation.LooperMode
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class TextInputActionTest {
+
+  @get:Rule
+  var activityRule =
+    ActivityScenarioRule<TextInputActionTestActivity>(
+      TextInputActionTestActivity.createIntent(ApplicationProvider.getApplicationContext())
+    )
+
+  lateinit var context: Context
+
+  @Before
+  fun setupContext() {
+    activityRule.scenario.onActivity {
+      context = it.baseContext
+    }
+    context.setTheme(R.style.Theme_AppCompat_Light)
```

Comment:
Is this still needed now that you're setting the theme directly in the test activity?

Reply:
Yes, removing this causes this error :
```java.lang.IllegalArgumentException: The style on this component requires your app theme to be Theme.AppCompat (or a descendant).```

Reply:
Hmm. The way this is being approached is not completely in-line with how ActivityScenario is meant to be used--we shouldn't be caching activity-tied objects at the test level since that can leak resources and cause unexpected failures.

What I'm wondering is whether you can construct the layout directly on the activity, e.g. (within each test):

````kotlin
activityRule.scenario.onActivity { activity ->
  val layout = TextInputLayout(activity)

  assert...
}
```

Since activity is already setting up its theme, the extra set theme is probably not needed. I suspect the reason it's needed here is because you're using context, not activity, later on (I'd expect removing the setTheme within the test activity to not cause a failure with the current code).

Reply:
This has been addressed in recent changes.